src/context/product-context.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+const ProductContext = createContext();
+
+const ProductContextProvider = ({ children }) => {
+  const axios = require("axios").default;
```

Comment:
Any specific reason behind importing `axios` here and not above?

Reply:
Nothing specific as such, i needed it inside provider so I imported here.